common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -1737,6 +1743,12 @@
     ],
     "sqlState" : "42822"
   },
+  "GROW_POINTER_ARRAY_OUT_OF_MEMORY" : {
```

Comment:
`GROW_POINTER_ARRAY_OUT_OF_MEMORY` -> `POINTER_ARRAY_OUT_OF_MEMORY`?

It's because the existing *_OUT_OF_MEMORY has the following naming format.

https://github.com/apache/spark/blob/921ef95609cde581d4ae32ab8ce18ec58913941d/common/utils/src/main/resources/error/error-conditions.json#L295

https://github.com/apache/spark/blob/921ef95609cde581d4ae32ab8ce18ec58913941d/common/utils/src/main/resources/error/error-conditions.json#L310

Reply:
thanks for suggestion, sounds reasonable, updated